internal/controller/rotators/aws_oidc_rotator.go

```diff
@@ -0,0 +1,199 @@
+// Copyright Envoy AI Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package rotators
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AWSOIDCRotator implements the Rotator interface for AWS OIDC token exchange.
+// It manages the lifecycle of temporary AWS credentials obtained through OIDC token
+// exchange with AWS STS.
+type AWSOIDCRotator struct {
+	// ctx provides a user specified context.
+	ctx context.Context
+	// client is used for Kubernetes API operations.
+	client client.Client
+	// kube provides additional Kubernetes API capabilities.
+	kube kubernetes.Interface
+	// logger is used for structured logging.
+	logger logr.Logger
+	// stsOps provides AWS STS operations interface.
+	stsOps STSClient
+	// backendSecurityPolicyName provides name of backend security policy.
+	backendSecurityPolicyName string
+	// backendSecurityPolicyNamespace provides namespace of backend security policy.
+	backendSecurityPolicyNamespace string
+	// preRotationWindow specifies how long before expiry to rotate.
+	preRotationWindow time.Duration
+}
+
+// NewAWSOIDCRotator creates a new AWS OIDC rotator with the specified configuration.
+// It initializes the AWS STS client and sets up the rotation channels.
+func NewAWSOIDCRotator(
+	ctx context.Context,
+	client client.Client,
+	kube kubernetes.Interface,
+	logger logr.Logger,
+	backendSecurityPolicyNamespace string,
+	backendSecurityPolicyName string,
+	preRotationWindow time.Duration,
+	region string,
+) (*AWSOIDCRotator, error) {
+	cfg, err := defaultAWSConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load AWS config: %w", err)
+	}
+
+	cfg.Region = region
+
+	if proxyURL := os.Getenv("AI_GATEWAY_STS_PROXY_URL"); proxyURL != "" {
```

Comment:
`AI_GATEWAY_STS_PROXY_URL` I guess we should document this somewhere. Maybe in the helm values?

Reply:
it is documented for podEnv in values.yaml